File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int total = 0;
        int j = mat.size() - 1;
        
        for (int i = 0; i < mat.size(); i++)
        {
            total += mat[i][i];
            
            if (i != j)
            {
                total += mat[j][i];
            }
            j--;
        }
        
        return total;
    }
};


            

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        sum = 0
        j = len(mat) - 1
        for i in range(len(mat)):
            sum += mat[i][i]
            if i != j:
                sum += mat[i][j]
            j -= 1
        
        return sum
            
#1